src/utils/change.link.ts

```diff
@@ -17,3 +19,15 @@ export const createSummaryListLink = (
 export const createChangeLinkConfig = (href: string, text: string, dataEventId: string) => {
   return createSummaryListLink('Change', href, text, dataEventId);
 };
+
+/**
+ * Return a change link with the transactionId and submissionId included
+ *   - In case of an exception, return an HTML-safe hash(#)
+ */
+export const createChangeLinkWithIds = (link: string, transactionId: string, submissionId: string) => {
+  try {
+    return getUrlWithTransactionIdAndSubmissionId(link, transactionId, submissionId);
+  } catch (e) {
+    return "#";
```

Comment:
I'm just interested in what kind of scenario an exception would be thrown (and caught here)? Also, what kind of URL would ultimately be used by the app/template? # followed by something else?

Reply:
The thinking behind this is that, in the very unlikely event that a url substitution fails, then we do not have to redirect the user to an error page and break the journey; the user can still view their `check-your-answers` page and see what they entered.

The `#` is an HTLM-safe option that does not break the mark-up. Obviously something would be wrong at this point that needs fixing (and would be reported by the user or tester), but the user would still be able to see a summary of their changes. Happy to discuss.

Reply:
We call the `getUrlWithTransactionIdAndSubmissionId()` function in quite a few places and it is invoked pretty much on every page when Redis flag is switched off. We haven't taken a defensive coding approach elsewhere so not convinced it's needed here either.

Reply:
Ok, no problem. I'll remove it. As long as you're happy that in case of a url substitution error/exception, it's better to terminate the journey and show the user an error screen, rather than have a view of their data so far entered.

Reply:
I'm happy. It's an unexpected scenario - something serious has gone wrong which might even mean that the data displayed on the 'check your answers' page isn't even correct. And like I said, the `getUrlWithTransactionIdAndSubmissionId()` function would have already been called in order to get them to this screen.

Reply:
Cheers Martin. Will go ahead and remove it.